### ISSUE TITLE
Can't set connectTimeout

### ISSUE THREAD

**Issue by wuxicn** [samuel/go-zookeeper#50]
Hi,

In zk/conn.go:145: `connectTimeout: 1 * time.Second,`
in very pool network condition or when DNS responses very slow, connection will fail.

I recommend to use `recvTimeout` as `connectTimeout` instead of Hard-code `1 * time.Second`.

**Comment by wuxicn** [comment]
I've fixed in: https://github.com/wuxicn/go-zookeeper/commit/028929e6ff8f5820d3cb6ed05b86d694716ce09d

**Comment by basvanbeek** [comment]
I don't think this is the right approach to solve the issue. You can already alter the connection timeout without a library change by providing your own Dialer function:

    myConnectTimeout := 5 * time.Second
    dialer := func(network, address string, _ time.Duration) (net.Conn, error) {
        return net.DialTimeout(network, address, myConnectTimeout)
    }
    conn, eventc, err := zk.Connect(servers, mySessionTimeout, zk.WithDialer(dialer))

I admit it is a bit verbose and maybe adding the ability in the library to alter the internal connection timeout through a connOption function would be better. I will add that to my already outstanding pull request.
